langserve/server.py

```diff
@@ -1190,193 +1110,514 @@ async def create_feedback(
             comment=feedback_from_langsmith.comment,
         )
 
-    #######################################
-    # Documentation variants of end points.
-    #######################################
-    # At the moment, we only support pydantic 1.x for documentation
-    if _PYDANTIC_MAJOR_VERSION == 1:
+    async def check_feedback_enabled(self, config_hash: str = "") -> None:
+        """Check if feedback is enabled for the runnable."""
+        if not tracing_is_enabled() or not self.enable_feedback_endpoint:
+            raise HTTPException(
+                400,
+                "The feedback endpoint is only accessible when LangSmith is "
+                + "enabled on your LangServe server.",
+            )
 
-        @app.post(
-            namespace + "/c/{config_hash}/invoke",
-            response_model=InvokeResponse,
-            tags=route_tags_with_config,
-            name=_route_name_with_config("invoke"),
-            description=(
-                "This endpoint is to be used with share links generated by the "
-                "LangServe playground. "
-                "The hash is an LZString compressed JSON string. "
-                "For regular use cases, use the /invoke endpoint without "
-                "the `c/{config_hash}` path parameter."
-            ),
+
+# PUBLIC API
+
+
+def add_routes(
+    app: Union[FastAPI, APIRouter],
+    runnable: Runnable,
+    *,
+    path: str = "",
+    input_type: Union[Type, Literal["auto"], BaseModel] = "auto",
+    output_type: Union[Type, Literal["auto"], BaseModel] = "auto",
+    config_keys: Sequence[str] = ("configurable",),
+    include_callback_events: bool = False,
+    enable_feedback_endpoint: bool = False,
+    per_req_config_modifier: Optional[PerRequestConfigModifier] = None,
+    with_invoke: bool = True,
```

Comment:
could we do something like 

`endpoint_blocklist: list[EndpointIdentifier] = []` 

and have a enum like `EndpointIdentifier.Batch`

Reply:
We can do that, though I think makes sense to use an allow list rather than a block list for a list -- the rationale being that if users pull a newer version of the package with extra endpoints, those endpoints won't be magically exposed

Reply:
I'm starting to think we should have a configuration object or something - so many arguments here...

Reply:
1) do you have a good example of when someone would want to use a block list (rather than allow list)?
2) the with_config_hash option doesn't correspond to any particular endpoint right now, but to config variants of all endpoints -- let me know if you have thoughts on how we should handle that, can still be a separate flag?